bitriseclient/bitriseclient.go

```diff
@@ -0,0 +1,237 @@
+package bitriseclient
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/retry"
+	"github.com/bitrise-io/go-utils/urlutil"
+)
+
+const (
+	baseURL                      = "https://api.bitrise.io/v0.1/"
+	appsEndPoint                 = "/apps"
+	provisioningProfilesEndPoint = "/provisioning-profiles"
+	certificatesEndPoint         = "/build-certificates"
+)
+
+// Paging ...
+type Paging struct {
+	TotalItemCount int    `json:"total_item_count"`
+	PageItemLimit  int    `json:"page_item_limit"`
+	Next           string `json:"next"`
+}
+
+// Owner ...
+type Owner struct {
+	AccountType string `json:"account_type"`
+	Name        string `json:"name"`
+	Slug        string `json:"slug"`
+}
+
+// Application ...
+type Application struct {
+	Slug        string `json:"slug"`
+	Title       string `json:"title"`
+	ProjectType string `json:"project_type"`
+	Provider    string `json:"provider"`
+	RepoOwner   string `json:"repo_owner"`
+	RepoURL     string `json:"repo_url"`
+	RepoSlug    string `json:"repo_slug"`
+	IsDisabled  bool   `json:"is_disabled"`
+	Status      int    `json:"status"`
+	IsPublic    bool   `json:"is_public"`
+	Owner       Owner  `json:"owner"`
+}
+
+// FetchMyAppsResponse ...
+type FetchMyAppsResponse struct {
+	Data   []Application `json:"data"`
+	Paging Paging        `json:"paging"`
+}
+
+// BitriseClient ...
+type BitriseClient struct {
+	accessToken     string
+	selectedAppSlug string
+	headers         map[string]string
+}
+
+// NewBitriseClient ...
+func NewBitriseClient(accessToken string) (client *BitriseClient, apps []Application, err error) {
```

Comment:
do not use named return values in this long functions, easy to shadow the `err`.
like in ln 108 you are checking the named return which is nil at that point!